### ISSUE TITLE
False positive in import plugin

### ISSUE THREAD

**Issue by Boshen** [oxc-project/oxc#3157]
These contain type exports

https://github.com/palantir/blueprint

```
blueprint  develop ❯ ~/github/oxc/target/release/oxlint --import-plugin -D export

  ⚠ eslint-plugin-import(export): No named exports found in module './itemRenderer'
    ╭─[packages/select/src/common/index.ts:21:15]
 20 │ export * from "./itemListRenderer";
 21 │ export * from "./itemRenderer";
    ·               ────────────────
 22 │ export * from "./listItemsProps";
    ╰────

  ⚠ eslint-plugin-import(export): No named exports found in module './predicate'
    ╭─[packages/select/src/common/index.ts:24:15]
 23 │ export * from "./listItemsUtils";
 24 │ export * from "./predicate";
    ·               ─────────────
 25 │ export type { SelectPopoverProps } from "./selectPopoverProps";
    ╰────

  ⚠ eslint-plugin-import(export): Multiple exports of name 'TimePrecision'.
    ╭─[packages/datetime/src/index.ts:29:10]
 28 │ export type { TimePickerProps } from "./common/timePickerProps";
 29 │ export { TimePrecision } from "./common/timePrecision";
    ·          ─────────────
 30 │
    ╰────
```

**Comment by Boshen** [comment]
```
blueprint  develop ❯ ~/github/oxc/target/release/oxlint --import-plugin -D named

  ⚠ eslint-plugin-import(named): named import "default" not found
   ╭─[packages/karma-build-scripts/index.mjs:7:10]
 6 │
 7 │ export { default as createKarmaConfig } from "./createKarmaConfig.mjs";
   ·          ───────
   ╰────
  help: does "./createKarmaConfig.mjs" have the export "default"?

  ⚠ eslint-plugin-import(named): named import "default" not found
    ╭─[packages/webpack-build-scripts/index.mjs:19:10]
 18 │ export { COMMON_EXTERNALS } from "./externals.mjs";
 19 │ export { default as baseConfig } from "./webpack.config.base.mjs";
    ·          ───────
 20 │ export { default as karmaConfig } from "./webpack.config.karma.mjs";
    ╰────
  help: does "./webpack.config.base.mjs" have the export "default"?

  ⚠ eslint-plugin-import(named): named import "default" not found
    ╭─[packages/webpack-build-scripts/index.mjs:20:10]
 19 │ export { default as baseConfig } from "./webpack.config.base.mjs";
 20 │ export { default as karmaConfig } from "./webpack.config.karma.mjs";
    ·          ───────
    ╰────
  help: does "./webpack.config.karma.mjs" have the export "default"?
```

**Comment by Boshen** [comment]
@Dunqing What should be the best way to fix this? Save all the export names for TypeScript types?

**Comment by Boshen** [comment]
`export type * from "mod"` doesn't fix the error 😅

**Comment by Dunqing** [comment]
> @Dunqing What should be the best way to fix this? Save all the export names for TypeScript types?

Yes, it seems we must. But after doing so we also need to deal with type merging. Such as allowing multiple namespaces with the same name, and allowing `export const A` and `export type A` with the same name.

**Comment by Dunqing** [comment]
> ```
>   ⚠ eslint-plugin-import(export): Multiple exports of name 'TimePrecision'.
>     ╭─[packages/datetime/src/index.ts:29:10]
>  28 │ export type { TimePickerProps } from "./common/timePickerProps";
>  29 │ export { TimePrecision } from "./common/timePrecision";
>     ·          ─────────────
>  30 │
>     ╰────
> ```

I double-checked this error and it's not a false positive. `TimePrecision` is also exported in common/index.ts. This file has `export * from ". /common"`

**Comment by Boshen** [comment]
> > ```
> >   ⚠ eslint-plugin-import(export): Multiple exports of name 'TimePrecision'.
> >     ╭─[packages/datetime/src/index.ts:29:10]
> >  28 │ export type { TimePickerProps } from "./common/timePickerProps";
> >  29 │ export { TimePrecision } from "./common/timePrecision";
> >     ·          ─────────────
> >  30 │
> >     ╰────
> > ```
> 
> I double-checked this error and it's not a false positive. `TimePrecision` is also exported in common/index.ts. This file has `export * from ". /common"`

How can we improve the error message 🤔

**Comment by Boshen** [comment]
> > @Dunqing What should be the best way to fix this? Save all the export names for TypeScript types?
> 
> Yes, it seems we must. But after doing so we also need to deal with type merging. Such as allowing multiple namespaces with the same name, and allowing `export const A` and `export type A` with the same name.

We can add a `exported_typings` to `ModuleRecord`.

**Comment by Boshen** [comment]
https://github.com/rolldown/rolldown/actions/runs/8968666129/job/24628561631?pr=1027

**Comment by Boshen** [comment]
@Dunqing I tested locally, they aren't fixed yet with ` --ignore-path=./.oxlintignore --import-plugin --deny-warnings` https://github.com/rolldown/rolldown/actions/runs/9011082137/job/24758169189?pr=1065
